cts.json

```diff
@@ -2115,6 +2115,160 @@
       "result": [
         {"d": "f"}
       ]
+    },
+    {
+      "name": "filter, length function, string data",
+      "selector" : "$[?length(@.a)>=2]",
+      "document" : [{"a": "ab"}, {"a": "d"}],
+      "result": [
+        {"a": "ab"}
+      ]
+    },
+    {
+      "name": "filter, length function, array data",
+      "selector" : "$[?length(@.a)>=2]",
+      "document" : [{"a": [1,2,3]}, {"a": [1]}],
+      "result": [
+        {"a": [1,2,3]}
+      ]
+    },
+    {
+      "name": "filter, length function, missing data",
+      "selector" : "$[?length(@.a)>=2]",
+      "document" : [{"d": "f"}],
+      "result": []
+    },
+    {
+      "name": "filter, length function, non-array/string arg",
+      "selector" : "$[?length(1)>=2]",
+      "invalid_selector": true
+    },
+    {
+      "name": "filter, length function, result must be compared",
+      "selector" : "$[?length(@.a)]",
+      "invalid_selector": true
+    },
+    {
+      "name": "filter, count function",
+      "selector" : "$[?count(@..*)>2]",
+      "document" : [
+        {"a": [1, 2, 3]},
+        {"a": [1], "d": "f"},
+        {"a": 1, "d": "f"}
+      ],
+      "result": [
+        {"a": [1,2,3]},
+        {"a": [1], "d":"f"}
+      ]
+    },
+    {
+      "name": "filter, count function, non-array/string arg",
+      "selector" : "$[?count(1)>2]",
+      "invalid_selector": true
+    },
+    {
+      "name": "filter, count function, result must be compared",
+      "selector" : "$[?count(@..*)]",
+      "invalid_selector": true
+    },
+    {
+      "name": "filter, match function, found match",
+      "selector" : "$[?match(@.a, '2022-01-[0-9]{2}')]",
+      "document" : [{"a": "2022-01-18"}],
+      "result": [
+        {"a":"2022-01-18"}
+      ]
+    },
+    {
+      "name": "filter, match function, don't select match",
+      "selector" : "$[?!match(@.a, '2022-01-[0-9]{2}')]",
```

Comment:
See https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/367 as to whether regexps may be enclosed in single quotes.

Reply:
I looked through the ABNF, and it looks like regex is just a string literal.

I also asked about it.  https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/362 is also releated.

Reply:
Depending on the outcome of #367, we should probably add double-quoted strings as a test case.

Reply:
Ah yes. Got it. Thanks.

Reply:
Added some double-quoted tests.